src/zos_files/zowe/zos_files_for_zowe_sdk/datasets.py

```diff
@@ -11,13 +11,216 @@
 """
 
 import os
+from typing import Optional
 
 from zowe.core_for_zowe_sdk import SdkApi
 from zowe.core_for_zowe_sdk.exceptions import FileNotFound
 from zowe.zos_files_for_zowe_sdk.constants import FileType, zos_file_constants
 
 _ZOWE_FILES_DEFAULT_ENCODING = zos_file_constants["ZoweFilesDefaultEncoding"]
 
+
+class DatasetOption:
+
+    def __init__(
+        self,
+        like: str = None,
+        volser: Optional[str] = None,
+        unit: Optional[str] = None,
+        dsorg: Optional[str] = None,
+        alcunit: Optional[str] = None,
+        primary: Optional[int] = None,
+        secondary: Optional[int] = None,
+        dirblk: Optional[int] = None,
+        avgblk: Optional[int] = None,
+        recfm: Optional[str] = None,
+        blksize: Optional[int] = None,
+        lrecl: Optional[int] = None,
+        storclass: Optional[str] = None,
+        mgntclass: Optional[str] = None,
+        dataclass: Optional[str] = None,
+        dsntype: Optional[str] = None,
+    ) -> None:
+        self.__like = like
+        self.__volser = volser
+        self.__unit = unit
+        self.dsorg = dsorg
+        self.alcunit = alcunit
+        self.primary = primary
+        self.secondary = secondary
+        self.__dirblk = dirblk
+        self.__avgblk = avgblk
+        self.recfm = recfm
+        self.blksize = blksize
+        self.__lrecl = lrecl
+        self.__storclass = storclass
+        self.__mgntclass = mgntclass
+        self.__dataclass = dataclass
+        self.__dsntype = dsntype
+
+    @property
+    def volser(self) -> Optional[str]:
+        return self.__volser
+
+    @volser.setter
+    def volser(self, volser: Optional[str]):
+        self.__volser = volser
+
+    @property
+    def unit(self) -> Optional[str]:
+        return self.__unit
+
+    @unit.setter
+    def unit(self, unit: Optional[str]):
+        self.__unit = unit
+
+    @property
+    def dsorg(self) -> Optional[str]:
+        return self.__dsorg
+
+    @dsorg.setter
+    def dsorg(self, dsorg: Optional[str]):
+        if dsorg not in ("PO", "PS"):
+            if self.like is not None and dsorg is None:
+                pass
+            else:
+                raise ValueError("")
+        self.__dsorg = dsorg
+
+    @property
+    def alcunit(self) -> Optional[str]:
+        return self.__alcunit
+
+    @alcunit.setter
+    def alcunit(self, alcunit: Optional[str]):
+        if alcunit is None:
+            if self.like is not None:
+                self.alcunit = None
+            else:
+                self.__alcunit = "TRK"
+        elif alcunit not in ("CYL", "TRK"):
+            raise KeyError
+        else:
+            self.__alcunit = alcunit
+
+    @property
+    def primary(self) -> Optional[int]:
+        return self.__primary
+
+    @primary.setter
+    def primary(self, primary: Optional[int]):
+        if primary is not None:
+            if primary > 16777215:
+                raise ValueError
```

Comment:
I have a similar suggestion here:

 ```suggestion
            if primary > 16777215:
                raise ValueError("Given primary space allocation exceeds track limit of 16,777,215")
```